core/slim_sim.cpp

```diff
@@ -7683,7 +7825,7 @@ void SLiMSim::__TallyMutationReferencesWithTreeSequence(std::unordered_map<slim_
 					int32_t allele_refs = 0;
 					
 					for (size_t sample_index = 0; sample_index < sample_count; sample_index++)
-						if ((variant->genotypes.u16[sample_index] == allele_index) && (indexToGenomeMap[sample_index] != nullptr))
+						if ((variant->genotypes.i16[sample_index] == allele_index) && (indexToGenomeMap[sample_index] != nullptr))
```

Comment:
please add a cast, `(int16_t)allele_index`, to get rid of a warning; allele_index is a `tsk_size_t` and the compiler is complaining that the types don't match

Reply:
WIll do. Weird though, mine isn't.

Reply:
Yeah, just compiling with different warning settings.  I build in Xcode, which I think has stricter settings than the cmake build uses.  Also uses a different compiler – clang instead of GCC.